File: go/search_in_a_binary_tree.go
package main

import "leet_code/tree"

// 700. Search in a Binary Search Tree
// You are given the root of a binary search tree (BST) and an integer val.

//   Find the node in the BST that the node's value equals val and return the subtree rooted with that node.
//  If such a node does not exist, return null.
func main()  {

}

func searchBST(root *tree.TreeNode, val int) *tree.TreeNode {
	if root == nil {
		return nil
	}
	if root.Val == val {
		return root
	}
	// 二叉树搜索树左子树节点值小于右子树节点值
	if root.Val > val {
		return searchBST(root.Left,val)
	}else{
		return searchBST(root.Right,val)
	}
}

File: php/search_in_a_binary_tree.php
<?php


// 700. Search in a Binary Search Tree
// You are given the root of a binary search tree (BST) and an integer val.

//   Find the node in the BST that the node's value equals val and return the subtree rooted with that node.
//  If such a node does not exist, return null.

//Definition for a binary tree node.
class TreeNode {
    public $val = null;
    public $left = null;
    public $right = null;
    function __construct($val = 0, $left = null, $right = null) {
        $this->val = $val;
        $this->left = $left;
        $this->right = $right;
    }
}

class Solution {

    /**
     * @param TreeNode $root
     * @param Integer $val
     * @return TreeNode
     */
    function searchBST($root, $val) {
        if($root == null){
            return null;
        }
        if($root->val == $val){
            return $root;
        }
        if($root->val > $val){
            // 去左边找
            return $this->searchBST($root->left,$val);
        }else{
            // 去右边找
            return $this->searchBST($root->right,$val);
        }
    }
}